# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoEncoder.h
# Commit: da42eb3f9664
# Full Hash: da42eb3f9664fc2d5dc77db2dc73e45b025c9a31
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Convert from MediaRawData to EncodedVideoChunk one by one. r=chunmin
#   
#   This simplifies the rest of the code, in remaining patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195713
# ==============================================================================

diff -r e59a8dc5add5 -r da42eb3f9664 dom/media/webcodecs/VideoEncoder.h
--- a/dom/media/webcodecs/VideoEncoder.h	Fri Dec 22 21:39:09 2023 +0000
+++ b/dom/media/webcodecs/VideoEncoder.h	Fri Dec 22 21:39:10 2023 +0000
@@ -64,12 +64,12 @@
       ErrorResult& aRv);
 
  protected:
-  virtual nsTArray<RefPtr<EncodedVideoChunk>> EncodedDataToOutputType(
-      nsIGlobalObject* aGlobal,
-      nsTArray<RefPtr<MediaRawData>>&& aData) override;
+  virtual RefPtr<EncodedVideoChunk> EncodedDataToOutputType(
+      nsIGlobalObject* aGlobal, RefPtr<MediaRawData>& aData) override;
 
   virtual VideoDecoderConfig EncoderConfigToDecoderConfig(
       nsIGlobalObject* aGlobal,
+      const RefPtr<MediaRawData>& aRawData,
       const VideoEncoderConfigInternal& mOutputConfig) const override;
 };
 
